app/controllers/annotations_controller.rb

```diff
@@ -30,8 +30,11 @@ def create
   # PUT /:course/annotations/1.json
   action_auth_level :update, :course_assistant
   def update
+    tweaked_params = annotation_params
+    tweaked_params.delete(:submission_id)
+    tweaked_params.delete(:filename)
```

Comment:
May I know why this is modified?

Reply:
From #1510: "Delete submission_id and filename to avoid association with the wrong submission"

Reply:
I believe that the intention was to ensure that when we update an annotation from a different submission than the submission it was originally created in, we do not accidentally override the original values of `submission_id` and `filename` with the new one.

But actually, this doesn't appear to be necessary; the `submission_id` and `filename` that is submitted via Ajax seems to always be the original values (because the data is derived from `@annotations` in `submissions/view.html`).